src/main/java/com/revature/services/QuestionLogicImpl.java

```diff
@@ -18,11 +24,100 @@
 	
 	@Autowired
 	private QuestionRepository dao;
+	
+	@Autowired
+	private QuestionEvalRepository qEvalDao;
+	
+	@Autowired
+	private QuestionCommentRepository commentDao;
+	
+	@Autowired
+	private SubjectRepository subjectDao;
+	
+	@Autowired
+	private EvalRepository evalDao;
+	
+	@PersistenceContext
+    private EntityManager entityManager;
+	
+//CREATE-----------------------------------
+	@Override
+	@Transactional
+	public QuestionPool createQuestion(QuestionPool question) {
+		
+		if (question.getMaxCommunicationScore() == null) {
+			throw new ConstraintViolationException("Missing required field maxCommunicationScore (Integer)", null);
+		}
+		
+		if (question.getMaxKnowledgeScore() == null) {
+			throw new ConstraintViolationException("Missing required field maxKnowledgeScore (Integer)", null);
+		}
+		
+		if (question.getQuestionText() == null) {
+			throw new ConstraintViolationException("Missing required field questionText (String)", null);
+		}
+		
+		if (question.getSubject() == null || question.getSubject().getId() == null) {
+			throw new ConstraintViolationException("Missing required field subject.id (Integer)", null);
+		}
+		
+		question.setUseCount(0);
+		question = dao.save(question);
+		// retrieve subject text - not auto filled
+		question.getSubject().setSubject(subjectDao.findOne(question.getSubject().getId()).getSubject());
+		return question;
+	}
 
 	@Override
+	@Transactional
+	public QuestionComment createComment(QuestionComment comment) {
+		return commentDao.save(comment);
+	}
+	
+	@Override
+	@Transactional
+	public QuestionComment createComment(QuestionComment comment, Integer questionId) {
+		comment.setQuestionEval(qEvalDao.findOne(questionId));
+		return commentDao.save(comment);
+	}
+	
+	@Override
+	@Transactional
+	public QuestionEval createQuestionEval(QuestionEval qEval, Integer evalId) {
+		
+		System.out.println(qEval);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS106)